enterprise/app/api_keys/api_keys.tsx

```diff
@@ -576,3 +575,64 @@ function newFormState<T extends ApiKeyFields>(request: T): FormState<T> {
     request,
   };
 }
+
+interface ApiKeyFieldProps {
+  value: string;
+}
+
+interface ApiKeyFieldState {
+  value: string;
+  isCopied: boolean;
+  hideValue: boolean;
+  displayValue: string;
+}
+
+const ApiKeyFieldDefaultState: ApiKeyFieldState = {
+  value: "",
+  isCopied: false,
+  hideValue: true,
+  displayValue: "************",
+};
+
+class ApiKeyField extends React.Component<ApiKeyFieldProps, ApiKeyFieldState> {
+  state = ApiKeyFieldDefaultState;
+
+  componentDidMount() {
+    this.setState({ value: this.props.value });
```

Comment:
since you're copying `this.props.value` directly to state and `this.state.value` is never modified, I think you could remove `value` from state and instead always reference `this.props.value`

Reply:
fixed